migrations/Version202203021737164734_taoBackOffice.php

```diff
@@ -0,0 +1,162 @@
+        'test/properties/test/timeLimits' => 'hide',
+        'delivery/taoLocal' => 'hide'
+    ];
+
+    const EXISTING_CONFIG_MAP = [
+        'tao' => [
+            'client_lib_config_registry' => [
+                'item/image/mediaAlignment' => [
+                    'ui/image/ImgStateActive' => ['mediaAlignment'],
+                ],
+                'test/item/BRS' => [
+                    'taoQtiTest/controller/creator/views/item' => ['BRS'],
+                ],
+                'item/commonInteractions/extendedTextInteraction/hasMath' => [
+                    'taoQtiItem/qtiCreator/widgets/interactions/extendedTextInteraction/states/Question' => ['hasMath'],
+                ],
+                'item/commonInteractions/hottextInteraction/disallowHTMLInHottext' => [
+                    'taoQtiItem/qtiCreator/widgets/interactions/hottextInteraction/states/Question'=> ['disallowHTMLInHottext'],
+                ]
+            ]
+        ],
+        'taoQtiItem' => [
+            'qtiCreator' => [
+                'item/multiColumn' => [
+                    'multi-column' => []
+                ],
+                'item/scrollableMultiColumn' => [
+                    'scrollable-multi-column' => []
+                ],
+                'item/perInteractionRp' => [
+                    'perInteractionRp' => []
+                ],
+            ]
+        ]
+    ];
+
+    public function up(Schema $schema): void
+    {
+        $resultConfig = self::DEFAULT_CONFIG;
+        $backOfficeExtension = \common_ext_ExtensionsManager::singleton()->getExtensionById('taoBackOffice');
+        if($backOfficeExtension === null) {
+            throw new \Exception(sprintf('Cannot find %s extension', 'taoBackOffice'));
+        }
+        $existingConfig = $backOfficeExtension->getConfig(self::GLOBAL_UI_CONFIG_NAME);
+        if($existingConfig !== false) {
+            $resultConfig = $existingConfig;
+        }
+
+        foreach(self::EXISTING_CONFIG_MAP as $extensionName => $extensionConfig) {
+            $currentExtension = \common_ext_ExtensionsManager::singleton()->getExtensionById($extensionName);
+            if($currentExtension === null) {
+                throw new \Exception(sprintf('Cannot find %s extension', $extensionName));
+            }
+            foreach($extensionConfig as $configFileName => $existingConfig) {
+                $currentExtensionConfig = $currentExtension->getConfig($configFileName);
+                if($currentExtensionConfig === false) {
+                    throw new \Exception(sprintf('Cannot find %s config', $configFileName));
+                }
+
+                $this->updateGlobalUIConfig($existingConfig, $currentExtensionConfig,$resultConfig);
+            }
+        }
+
+        $backOfficeExtension->setConfig(self::GLOBAL_UI_CONFIG_NAME, $resultConfig);
+    }
+
+    private function updateGlobalUIConfig($existingConfig, $currentExtConfig, array &$resultConfig): void
```

Comment:
Please, avoid changing variables by reference. It's usually harder to debug and maintain. I'd recommend returning the new config instead.

Reply:
fixed